Projects/Server/Utilities/Utility.cs

```diff
@@ -972,20 +972,20 @@ public static TimeSpan GetXMLTimeSpan(string timeSpanString, TimeSpan defaultVal
     public static int Dice(int numDice, int numSides, int bonus)
     {
       int total = 0;
+      uint sides = (uint)numSides;

```

Comment:
Before the cast do an if check that it's greater than or equal to 0, or change the API to use a uint as it should have been.

I always assume that there are lots of badly written scripts out there full of bugs calling these functions. Before, passing a negative value would have just given a give roll of 0. This cast would change it to possibly be a very large value.

Reply:
I changed it to a uint and then changed it back. I don't remember why, but you are right.